omics/analyzer/omics-run-analyzer

```diff
@@ -0,0 +1,460 @@
+            if not stop1 or stop > stop1:
+                stop1 = stop
+        else:
+            stops.append(idx)
+    for idx in stops:
+        events.append({"time": stop1, "event": "stop", "index": idx})
+    events.sort(key=lambda x: x["time"])
+
+    metric_names = [
+        "cpusReserved",
+        "cpusMaximum",
+        "cpusAverage",
+        "gpusReserved",
+        "memoryReservedGiB",
+        "memoryMaximumGiB",
+        "memoryAverageGiB",
+    ]
+    metrics = run.get("metrics", {})
+    run["metrics"] = metrics
+
+    active = []
+    t0 = None
+    time = 0
+    for evt in events:
+        t1 = evt["time"]
+        if t0:
+            secs = (t1 - t0).total_seconds()
+            time += secs
+            for name in metric_names:
+                task_metrics = tasks[idx].get("metrics", {})
+                mvalues = [task_metrics.get(name) for idx in active]
+                mvalues = [v for v in mvalues if v is not None]
+                if not mvalues:
+                    continue
+                total = sum(mvalues)
+                if "Average" in name:
+                    metrics[name] = metrics.get(name, 0) + total * secs
+                else:
+                    metrics[name] = max(metrics.get(name, total), total)
+        t0 = t1
+        if evt["event"] == "start":
+            active.append(evt["index"])
+        elif evt["index"] in active:
+            active.remove(evt["index"])
+
+    for name in metric_names:
+        if name in metrics and "Average" in name:
+            metrics[name] /= time
+
+
+def add_metrics(res, resources, pricing):
+    """Add run/task metrics"""
+    metrics = res.get("metrics", {})
+    res["metrics"] = metrics
+
+    arn = re.split(r"[:/]", res["arn"])
+    rtype = arn[-2]
+    region = arn[3]
+    res["type"] = rtype
+    if rtype == "run":
+        add_run_util(res, resources[1:])
+
+    time1 = parse_time_str(res.get("startTime"))
+    time2 = parse_time_str(res.get("stopTime"))
+    running = 0
+    if time1 and time2:
+        running = (time2 - time1).total_seconds()
+        metrics["runningSeconds"] = running
+
+    cpus_res = metrics.get("cpusReserved")
+    cpus_max = metrics.get("cpusMaximum")
+    if cpus_res and cpus_max:
+        metrics["cpuUtilization"] = float(cpus_max) / float(cpus_res)
+    gpus_res = metrics.get("gpusReserved")
+    mem_res = metrics.get("memoryReservedGiB")
+    mem_max = metrics.get("memoryMaximumGiB")
+    if mem_res and mem_max:
+        metrics["memoryUtilization"] = float(mem_max) / float(mem_res)
+    store_res = metrics.get("storageReservedGiB")
+    store_max = metrics.get("storageMaximumGiB")
+    if store_res and store_max:
+        metrics["storageUtilization"] = float(store_max) / float(store_res)
+
+    if rtype == "run":
+        capacity = get_storage_gib(res.get("storageCapacity"))
+        metrics["sizeReserved"] = f"{capacity} GiB"
+        gib_hrs = capacity * running / 3600.0
```

Comment:
worth stashing the value of `3600.0` as a constant of `SEC_PER_HOUR`?

Reply:
Sure